evalml/data_checks/id_columns_data_check.py

```diff
@@ -189,49 +204,53 @@ def validate(self, X, y=None):
 
         first_col_id = False
 
-        if col_names and col_names[0] in id_cols_above_threshold:
+        if (
+            col_names
+            and col_names[0] in id_cols_above_threshold
+            and id_cols_above_threshold[col_names[0]] == 1.0
+        ):
             first_col_id = True
```

Comment:
would it make sense to just do the warning append in this conditional block to keep all that logic together?

Reply:
Yeah you're right, made a change to remove the redundant logic